### ISSUE TITLE
Error 1556 when accessing GOOP4 passed to Actor

### ISSUE THREAD

**Issue by pettaa123** [opengds/OpenGDS#116]
Hello,
I am not sure if this code have ever worked but I get error 1556 when trying to access a GOOP4 class' attribute from an Actor, when the vi holding the Create finishes before I read in Actor Core.
![image](https://github.com/user-attachments/assets/17e52d2e-2ddd-41bd-9332-4ed9658392fd)
I am using LV 2023Q4 , but I don't think this: https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000001EABFCA4&l=en-GB
is related, since the application instance looks the same for all VIs IMHO.

**Comment by achuthaperumal** [comment]
@pettaa123 Did you create the DVR_Actor with GOOP Suite? or with Default Actor Script? Can you share the `Write Test.vi`?

**Comment by pettaa123** [comment]
This is a defualt NI's Actor. So the DVR goes out of scope and gets deleted when Untitled 1.vi finishes execution.
Do you know how to inherit from MGI's Panel Actor instead of Actor? When adding Panel Actor.lvlib to my project, Panel Actor doesn't show up as option.
![image](https://github.com/user-attachments/assets/c2c8da63-9079-4ac2-99f3-64b7ebfd9d0b)

**Comment by pettaa123** [comment]
However, I created a pure GOOP4 Actor with mObj ObjectAttribute member.
But when creating mObj in a VI before setting it to GOOP4 Actor, the mObj DVR gets deleted, despite added to the Actor.
![image](https://github.com/user-attachments/assets/eb772b4c-2623-4595-839c-cd692be28b3a)

**Comment by achuthaperumal** [comment]
@pettaa123 May I know why are you calling the `_Create.vi` twice? That's the problem! When you call the `_Create.vi` in the `Actor Core.vi` again, It creates a new DVR and the old DVR will be overwritten.

Make Sure the `_Create.vi` is called only once per instance.

**Comment by achuthaperumal** [comment]
> This is a defualt NI's Actor. So the DVR goes out of scope and gets deleted when Untitled 1.vi finishes execution. Do you know how to inherit from MGI's Panel Actor instead of Actor? When adding Panel Actor.lvlib to my project, Panel Actor doesn't show up as option. ![image](https://private-user-images.githubusercontent.com/31046837/368310430-c2c8da63-9079-4ac2-99f3-64b7ebfd9d0b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2MzYxMzYsIm5iZiI6MTcyNjYzNTgzNiwicGF0aCI6Ii8zMTA0NjgzNy8zNjgzMTA0MzAtYzJjOGRhNjMtOTA3OS00YWMyLTk5ZjMtNjRiN2ViZmQ5ZDBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE4VDA1MDM1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE4MjA4MzMyM2U0ZTVlOGNkN2I5MTYyNDNiZjVhNzJkMTUxODc0ZTBhNGM1ZThiZjNjNTNlNTI5NmFlNWEyNTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pLwYS7dZ9VdTiIcfBS_lcZjC_golpHT83PIfmrDbCbI)

Currently, I'm exploring the code to see if there's support to MGI Panel Actor. I shall let you know if there's any way.

**Comment by pettaa123** [comment]
I think for Actor's it is advantageous to init their members before launching them. This makes error handling easier. It seems this is impossible right now. Maybe it is better to use GOOP3 with actors then?

**Comment by achuthaperumal** [comment]
> I think for Actor's it is advantageous to init their members before launching them. This makes error handling easier. It seems this is impossible right now, at least with GOOP4 members. Maybe it is better use GOOP3 members with actors then?

Hi, It's possible with GOOP4. But you need to delete the `_Create.vi` once the Actor is generated.
